File: benchs/gcd/gcd.ml
let gcd m n =
  let rec loop a b = 
    if a = b then a
    else if a < b then loop a (b-a)
    else loop (a-b) b in
  loop m n

let rec repeat n =
  if n <= 0 then () else (
    ignore (gcd 500001 500003);
    (repeat(n-1)))

let main () = 
  let r = repeat 20 in r;
;;

main() ;;

File: benchs/gcd/gcd.py
from microbit import *

def gcd(m,n):
    while (m !=  n): 
        if (m < n):
            n = n-m
        else:
            m =  m-n 
    return m

def repeat(n):
    r = 0
    i = 0 
    while (i< n):
        r=r+gcd(500001,500003)
        i = i + 1
    return r

def go(n):
    repeat(n)

display.show(Image.CONFUSED)
go(10)
display.show(Image.HAPPY)
